lib/active_model_serializers/logging.rb

```diff
@@ -0,0 +1,118 @@
+##
+# ActiveModelSerializers::Logging
+#
+#   https://github.com/rails/rails/blob/280654ef88/activejob/lib/active_job/logging.rb
+#
+module ActiveModelSerializers
+  module Logging
+    extend ActiveSupport::Concern
+
+    included do
+      include ActiveModelSerializers::Callbacks
+      extend Macros
+      instrument_rendering
+    end
+
+    module ClassMethods
+      def instrument_rendering
+        around_render do |args, block|
+          tag_logger do
+            notify_render do
+              block.call(args)
+            end
+          end
+        end
+      end
+    end
+
+    # Macros that can be used to customize the logging of class or instance methods,
+    # by extending the class or its singleton.
+    #
+    # Adapted from:
+    #   https://github.com/rubygems/rubygems/blob/cb28f5e991/lib/rubygems/deprecate.rb
+    #
+    # Provides a single method +notify+ to be used to declare when
+    # something a method notifies, with the argument +callback_name+ of the notification callback.
+    #
+    #     class Adapter
+    #       def self.klass_method
+    #         # ...
+    #       end
+    #
+    #       def instance_method
+    #         # ...
+    #       end
+    #
+    #       include ActiveModelSerializers::Logging::Macros
+    #       notify :instance_method, :render
+    #
+    #       class << self
+    #         extend ActiveModelSerializers::Logging::Macros
+    #         notify :klass_method, :render
+    #       end
+    #     end
+    module Macros
+      ##
+      # Simple notify method that wraps up +name+
+      # in a dummy method. It notifies on with the +callback_name+ notifier on
+      # each call to the dummy method, telling what the current serializer and adapter
+      # are being rendered.
+      # Adapted from:
+      #   https://github.com/rubygems/rubygems/blob/cb28f5e991/lib/rubygems/deprecate.rb
+      def notify(name, callback_name)
+        class_eval do
+          old = "_notifying_#{callback_name}_#{name}"
+          alias_method old, name
+          define_method name do |*args, &block|
+            run_callbacks callback_name do
+              send old, *args, &block
+            end
+          end
+        end
+      end
+    end
+
+    def notify_render(*)
+      event_name = 'render.active_model_serializers'.freeze
+      ActiveSupport::Notifications.instrument(event_name, notify_render_payload) do
+        yield
+      end
+    end
+
+    def notify_render_payload
+      { serializer: serializer, adapter: adapter }
+    end
+
+    private
+
+    def tag_logger(*tags)
+      if ActiveModelSerializers.logger.respond_to?(:tagged)
+        tags.unshift 'AMS'.freeze unless logger_tagged_by_active_model_serializers?
```

Comment:
I'd consider it... Or even rename it to activemodelserializers... This is pretty simple and in line w rails and activejob

B mobile phone

> On Nov 10, 2015, at 8:26 AM, L. Preston Sego III <notifications@github.com> wrote:
> 
> In lib/active_model_serializers/logging.rb:
> 
> > +    def notify_render(*)
> > +      event_name = 'render.active_model_serializers'.freeze
> > +      ActiveSupport::Notifications.instrument(event_name, notify_render_payload) do
> > +        yield
> > +      end
> > +    end
> > +
> > +    def notify_render_payload
> > +      { serializer: serializer, adapter: adapter }
> > +    end
> > +
> > +    private
> > +
> > +    def tag_logger(*tags)
> > +      if ActiveModelSerializers.logger.respond_to?(:tagged)
> > +        tags.unshift 'AMS'.freeze unless logger_tagged_by_active_model_serializers?
> should 'AMS'.freeze be a constant somewhere? I see it all over the place
> 
> —
> Reply to this email directly or view it on GitHub.
>